controllers/openstackcluster_controller.go

```diff
@@ -181,6 +181,10 @@ func reconcileDelete(ctx context.Context, log logr.Logger, client client.Client,
 	}
 
 	// Delete other things
+	if err = networkingService.DeleteOrphanedPorts(); err != nil {
```

Comment:
This should fix: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/issues/723